src/applications/personalization/profile-2/tests/e2e/profile.personal-and-contact-information-modals.cypress.spec.js

```diff
@@ -0,0 +1,176 @@
+import { PROFILE_PATHS } from '../../constants';
+
+import mockUser from '../fixtures/users/user-36.json';
+import mockFeatureToggles from '../fixtures/feature-toggles.json';
+
+const setup = (mobile = false) => {
+  window.localStorage.setItem(
+    'DISMISSED_ANNOUNCEMENTS',
+    JSON.stringify(['single-sign-on-intro']),
+  );
+
+  if (mobile) {
+    cy.viewport('iphone-4');
+  }
+
+  cy.login(mockUser);
+  cy.route('GET', '/v0/feature_toggles*', mockFeatureToggles);
+  cy.visit(PROFILE_PATHS.PROFILE_ROOT);
+
+  // should show a loading indicator
+  cy.findByRole('progressbar').should('exist');
+  cy.findByText(/loading your information/i).should('exist');
+
+  // and then the loading indicator should be removed
+  cy.findByText(/loading your information/i).should('not.exist');
+  cy.findByRole('progressbar').should('not.exist');
+};
+
+const checkModals = options => {
+  const {
+    editButtonId,
+    otherSectionEditButtonId,
+    editLineId,
+    sectionName,
+  } = options;
+
+  // Open edit view
+  cy.get(`#${editButtonId}-edit-link`).click({
```

Comment:
is it much slower to do:
```
getByRole('button', {name: `edit ${sectionName}` exact: false})
```
to get the edit button?

Reply:
would have to do this, and it doesn't seem much slower
```
cy.findByRole('button', {
    name: new RegExp(`edit ${sectionName}`, 'i'),
  }).click({
    force: true,
  });
```

Reply:
Updated! 💯